nightfall-deployer/migrations/2_deploy_upgradeable.js

```diff
@@ -10,11 +10,11 @@ const Proposers = artifacts.require('Proposers.sol');
 const Challenges = artifacts.require('Challenges.sol');
 const State = artifacts.require('State.sol');
 const SimpleMultiSig = artifacts.require('SimpleMultiSig.sol');
-const KYC = artifacts.require('KYC.sol');
+const X509 = artifacts.require('X509.sol');
 
 const config = require('config');
 
-const { RESTRICTIONS, MULTISIG, WHITELIST_MANAGERS } = config;
+const { RESTRICTIONS, MULTISIG, WHITELIST_MANAGERS, RSA_TRUST_ROOTS } = config;
```

Comment:
I understand RSA_TRUST_ROOTS somehow initializes the chain of trust between X509 contract and end user. My problem is that I fail how this RSA_TRUST_ROOTS is to be initialized in production.

Reply:
In production RSA_TRUST_ROOTS, would contain the root public keys of the CAs that we have decided to trust ( Digicert, Verisign etc.).  These are the public keys in their self-signed root certificates and they're publicly available.  For example Digicert's is [here](https://www.digicert.com/kb/digicert-root-certificates.htm).  `Owner` can also add remove keys after deployment (although we need to add this functionality to the admin container).
For now, I just made my own CA (self-signed cert) and told the contract to trust it, then made an intermediate CA certificate from that and finally an end user certificate.  At some point soon, we should test with an end-user cert derived from a real CA but that is a little involved (identity checks etc) and so, in the interestsof time, I haven't done that here.